### PR TITLE
Add log_to option: logfile

### PR DESCRIPTION
**Pull request by coderberry**
First, let me thank you for this gem. It's awesome!

I added the option to send the logs to a specific `wrapped_print.log` file by using the `log_to: :logfile`. This is very useful for me and perhaps others will appreciate it as well.

Thanks!

Eric

### CODE REVIEW

File: lib/wrapped_print.rb [hunk 1]
```diff
@@ -51,8 +52,12 @@ def wp(label = nil, pattern: PATTERN, count: COUNT, prefix: nil, suffix: nil, co
 
     private
     def detect_logger_method
-      if WrappedPrint.log_to == :logs
+      case WrappedPrint.log_to
+      when :logs
         Rails.logger.method(WrappedPrint.level)
+      when :logfile
+        @logfile ||= Logger.new("#{Rails.root}/log/wrapped_print.log")
```
Comment (igorkasyanchuk) [inline lib/wrapped_print.rb]: can you make it configurable? I mean filename.

also 2 simple specs just to cover this method that it just works!

**Comment by igorkasyanchuk** [comment]
thank you for feedback, I have left a comment. I think if you add a quick improvement and specs I'll merge ASAP

**Comment by coderberry** [comment]
@igorkasyanchuk Thanks for the feedback. Please let me know if my changes look ok or if you'd like any modifications.

File: lib/wrapped_print.rb [hunk 2]
```diff
@@ -14,6 +18,9 @@ module WrappedPrint
   mattr_accessor :log_to
   @@log_to = :console
 
+  mattr_accessor :custom_logger
```
Comment (igorkasyanchuk) [inline lib/wrapped_print.rb]: I think you can use log_to variable

I mean if you want to log to file

```
    WrappedPrint.setup do |config|
      config.log_to = ActiveSupport::Logger.new("some/file.log")
    end
```

no need to specify "custom_logger" and have a fallback to  ActiveSupport::Logger.new("log/wrapped_print.log")

File: lib/wrapped_print.rb [hunk 3]
```diff
@@ -51,8 +58,14 @@ def wp(label = nil, pattern: PATTERN, count: COUNT, prefix: nil, suffix: nil, co
 
     private
     def detect_logger_method
-      if WrappedPrint.log_to == :logs
+      case WrappedPrint.log_to
+      when :logs
```
Comment (igorkasyanchuk) [inline lib/wrapped_print.rb:58]: here you can do something like

```ruby
return WrappedPrint.logger.method(WrappedPrint.level) if WrappedPrint.logger.is_a?(ActiveSupport::Logger)
```

File: spec/wrapped_print_spec.rb [hunk 4]
```diff
@@ -14,6 +14,25 @@ def balance
     expect(WrappedPrint::VERSION).not_to be nil
   end
 
+  it "writes to a custom logger" do
+    logger_io = StringIO.new
+
+    WrappedPrint.setup do |config|
```
Comment (igorkasyanchuk) [inline spec/wrapped_print_spec.rb:20]: config.log_to = ActiveSupport::Logger.new(logger_io)

**Comment by igorkasyanchuk** [comment]
@coderberry please few more improvements :) I think tomorrow it will be ready for merge

**Comment by coderberry** [comment]
@igorkasyanchuk I have made the requested changes. Thank you for your guidance. My editor auto-cleaned the extra spaces in the README. I can undo those if you'd like.

**Comment by igorkasyanchuk** [comment]
merged 👍 will release it right now

**Comment by igorkasyanchuk** [comment]
wrapped_print-0.1.1.gem

**Pull request merged**